app/controllers/tickets_controller.rb

```diff
@@ -19,4 +19,21 @@ def create
 
     redirect_to root_path
   end
+
+  before_action :authenticate_user!
```

Comment:
`before_action :authenticate_user!` które jest w linijce 9 działa dla całego kontrolera. 
Jeśli chce się je ograniczyć do wybranych metod to trzeba napisać: `before_action :authenticate_user!, only: [:index, :buy]`